pkg/build/api/v1beta3/types.go

```diff
@@ -382,6 +386,21 @@ type SourceBuildStrategy struct {
 	ForcePull bool `json:"forcePull,omitempty" description:"forces the source build to pull the image if true"`
 }
 
+// BuildPostCommitSpec holds a build post commit hook specification.
+type BuildPostCommitSpec struct {
+	// Shell is an action that executes a shell script in a container
+	// running the build output image.
+	Shell string `json:"shell,omitempty" description:"shell script to be executed in a container running the build output image"`
```

Comment:
The proposal was not to concatenate with spaces - the proposal was to do proper quoting for the user.  My primary concern is that the assumption that users *don't* need the assistance to do arguments properly.  Shell escaping is extremely difficult to get correct in general, and an end user who *wants* proper shell escaping may not have the technical expertise in bash to express their intent correctly.  That's what made shellshock so dangerous - no one understands bash quoting, therefore everyone was potentially at risk.

A user who wants to properly separate args in a command execution has to fully understand bash argument syntax to pass shell as a single string.  That does not describe 90% of our users - who come from Windows backgrounds.  Taking an array of shell arguments is an *affordance* to their expectations - it sets them up for success, not for obscure bugs.